src/sentry/integrations/jira_server/client.py

```diff
@@ -95,40 +101,33 @@ def request(self, *args, **kwargs):
         return self._request(*args, **kwargs)
 
 
-class JiraServerClient(ApiClient):
+class JiraServer(object):
     """
-    Client for making authenticated requests to JiraServer
+    Contains the jira-server specifics that a JiraClient needs
+    in order to communicate with jira
     """
 
-    def __init__(self, installation):
-        self.installation = installation
-        super(JiraServerClient, self).__init__(self.metadata['verify_ssl'])
-        self.base_url = self.metadata['base_url']
+    def __init__(self, credentials):
+        self.credentials = credentials
 
     @property
-    def identity(self):
-        return self.installation.default_identity
+    def cache_prefix(self):
+        return 'sentry-jira-server:'
 
-    @property
-    def metadata(self):
-        return self.installation.model.metadata
-
-    def request(self, *args, **kwargs):
+    def request_hook(self, method, path, params, **kwargs):
+        """
+        Used by Jira Client to apply the jira-server authentication
+        Which is RSA signed OAuth1
+        """
         if 'auth' not in kwargs:
-            auth_data = self.identity.data
             kwargs['auth'] = OAuth1(
-                client_key=auth_data['consumer_key'],
-                rsa_key=auth_data['private_key'],
-                resource_owner_key=auth_data['access_token'],
-                resource_owner_secret=auth_data['access_token_secret'],
+                client_key=self.credentials['consumer_key'],
+                rsa_key=self.credentials['private_key'],
+                resource_owner_key=self.credentials['access_token'],
+                resource_owner_secret=self.credentials['access_token_secret'],
                 signature_method=SIGNATURE_RSA,
                 signature_type='auth_header')
-        return self._request(*args, **kwargs)
-
-    def get_valid_statuses(self):
-        # TODO Implement this.
-        return []
 
-    def get_projects_list(self):
-        # TODO Implement this
-        return []
+        request_spec = kwargs.copy()
```

Comment:
Just curious why is `kwargs` copied here?

Reply:
I was concerned about mutating the dictionary in the caller. Python dicts pass by reference so cloning felt safer. Example of this is

```python
>>> d = {}
>>> def mut(a):
...   a['key'] = 'x'
...
>>> mut(d)
>>> d
{'key': 'x'}
```